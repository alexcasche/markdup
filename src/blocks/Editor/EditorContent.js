import PreviewContent from '../Preview/PreviewContent'

const EditorContent = PreviewContent.extend`
  padding: ${props => (props.modifiers.indexOf('expanded') === -1 ? '0' : '')};
`

export default EditorContent
