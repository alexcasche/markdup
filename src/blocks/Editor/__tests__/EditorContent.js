import EditorContent from '../EditorContent'

describe('EditorContent Element', () => {
  it('renders correctly', () => {
    const tree = render(<EditorContent modifiers={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it('adds expanded modifiers', () => {
  const tree = render(<EditorContent modifiers={['expanded']} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('padding-bottom', '60px')
})
