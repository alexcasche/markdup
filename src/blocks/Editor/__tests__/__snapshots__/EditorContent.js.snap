// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorContent Element renders correctly 1`] = `
.c0 {
  height: calc(100% - 46px);
  padding: 15px;
  overflow-y: scroll;
  word-wrap: break-word;
  color: #1D1F21;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.5;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  padding: 0;
}

.c0::-webkit-scrollbar {
  width: 15px;
  position: absolute;
}

.c0::-webkit-scrollbar-track {
  background-color: #f7f7f7;
}

.c0::-webkit-scrollbar-thumb {
  background-color: #e1e2e4;
}

.c0::-webkit-scrollbar-thumb:hover {
  background-color: #d5d6d9;
}

.c0 > div > *:first-child {
  margin-top: 0;
}

.c0 h1 {
  padding-bottom: 0.3em;
  font-size: 32px;
  border-bottom: 1px solid #eaecef;
  font-weight: 500;
  line-height: 1.25;
  margin: 0.67em 0;
}

.c0 h2 {
  padding-bottom: 0.3em;
  font-size: 1.5em;
  border-bottom: 1px solid #eaecef;
  margin-top: 24px;
  margin-bottom: 16px;
  font-weight: 500;
  line-height: 1.25;
}

.c0 p {
  margin-top: 0;
  margin-bottom: 16px;
  line-height: 1.5;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Helvetica, Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji', 'Segoe UI Symbol';
}

.c0 a {
  color: #0366d6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 a:hover,
.c0 a:active {
  outline-width: 0;
}

.c0 ul {
  padding-left: 2em;
  margin-top: 0;
  margin-bottom: 16px;
}

.c0 ul li + li {
  margin-top: 0.25em;
}

.c0 pre {
  padding: 16px;
  overflow: auto;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
  font-size: 16px;
}

.c0 code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  font-family: 'SFMono-Regular',Consolas,'Liberation Mono',Menlo,Courier, monospace;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c0 table {
  display: block;
  width: 100%;
  overflow: auto;
  margin-top: 0;
  margin-bottom: 16px;
  border-spacing: 0;
  border-collapse: collapse;
}

.c0 tr {
  background-color: #fff;
  border-top: 1px solid #c6cbd1;
}

.c0 tr:nth-child(2n) {
  background-color: #f6f8fa;
}

.c0 th {
  text-align: center;
  font-weight: 600;
}

.c0 th,
.c0 td {
  padding: 6px 13px;
  border: 1px solid #dfe2e5;
}

.c0 table code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

<div
  className="c0"
/>
`;

exports[`adds expanded modifiers 1`] = `
.c0 {
  height: calc(100% - 46px);
  padding: 15px;
  overflow-y: scroll;
  word-wrap: break-word;
  color: #1D1F21;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.5;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  height: calc(100% - 46px);
  margin-top: 46px;
  padding-top: 34px;
  padding-right: calc(10vw - 15px);
  padding-bottom: 60px;
  padding-left: 10vw;
}

.c0::-webkit-scrollbar {
  width: 15px;
  position: absolute;
}

.c0::-webkit-scrollbar-track {
  background-color: #f7f7f7;
}

.c0::-webkit-scrollbar-thumb {
  background-color: #e1e2e4;
}

.c0::-webkit-scrollbar-thumb:hover {
  background-color: #d5d6d9;
}

.c0 > div > *:first-child {
  margin-top: 0;
}

.c0 h1 {
  padding-bottom: 0.3em;
  font-size: 32px;
  border-bottom: 1px solid #eaecef;
  font-weight: 500;
  line-height: 1.25;
  margin: 0.67em 0;
}

.c0 h2 {
  padding-bottom: 0.3em;
  font-size: 1.5em;
  border-bottom: 1px solid #eaecef;
  margin-top: 24px;
  margin-bottom: 16px;
  font-weight: 500;
  line-height: 1.25;
}

.c0 p {
  margin-top: 0;
  margin-bottom: 16px;
  line-height: 1.5;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Helvetica, Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji', 'Segoe UI Symbol';
}

.c0 a {
  color: #0366d6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 a:hover,
.c0 a:active {
  outline-width: 0;
}

.c0 ul {
  padding-left: 2em;
  margin-top: 0;
  margin-bottom: 16px;
}

.c0 ul li + li {
  margin-top: 0.25em;
}

.c0 pre {
  padding: 16px;
  overflow: auto;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
  font-size: 16px;
}

.c0 code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  font-family: 'SFMono-Regular',Consolas,'Liberation Mono',Menlo,Courier, monospace;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c0 table {
  display: block;
  width: 100%;
  overflow: auto;
  margin-top: 0;
  margin-bottom: 16px;
  border-spacing: 0;
  border-collapse: collapse;
}

.c0 tr {
  background-color: #fff;
  border-top: 1px solid #c6cbd1;
}

.c0 tr:nth-child(2n) {
  background-color: #f6f8fa;
}

.c0 th {
  text-align: center;
  font-weight: 600;
}

.c0 th,
.c0 td {
  padding: 6px 13px;
  border: 1px solid #dfe2e5;
}

.c0 table code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

<div
  className="c0"
/>
`;
